pkg/config/s3.go

```diff
@@ -2,11 +2,12 @@ package config
 
 // S3Config specifies the properties required to use S3 as the storage backend
 type S3Config struct {
-	Region                  string `validate:"required" envconfig:"AWS_REGION"`
-	Key                     string `envconfig:"AWS_ACCESS_KEY_ID"`
-	Secret                  string `envconfig:"AWS_SECRET_ACCESS_KEY"`
-	Token                   string `envconfig:"AWS_SESSION_TOKEN"`
-	Bucket                  string `validate:"required" envconfig:"ATHENS_S3_BUCKET_NAME"`
-	UseDefaultConfiguration bool   `envconfig:"AWS_USE_DEFAULT_CONFIGURATION"`
-	CredentialsEndpoint     string `envconfig:"AWS_CREDENTIALS_ENDPOINT"`
+	Region                             string `validate:"required" envconfig:"AWS_REGION"`
+	Key                                string `envconfig:"AWS_ACCESS_KEY_ID"`
+	Secret                             string `envconfig:"AWS_SECRET_ACCESS_KEY"`
+	Token                              string `envconfig:"AWS_SESSION_TOKEN"`
+	Bucket                             string `validate:"required" envconfig:"ATHENS_S3_BUCKET_NAME"`
+	UseDefaultConfiguration            bool   `envconfig:"AWS_USE_DEFAULT_CONFIGURATION"`
+	CredentialsEndpoint                string `envconfig:"AWS_CREDENTIALS_ENDPOINT"`
+	AwsContainerCredentialsRelativeURI string `envconfig:"AWS_CONTAINER_CREDENTIALS_RELATIVE_URI"`
```

Comment:
@manugupt1: I think if `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI` exists, athens should automatically set `CredentialsEndpoint` to `http://169.254.170.2` (this is the Fargate endpoint host for all services, it never changes).

For example, let's say:
* `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI ` = `/v2/credentials/e7e38a7b-8ed8-4245-8557-d88b02aca7c7`

Then make `AWS_CREDENTIALS_ENDPOINT` = `http://169.254.170.2` and `endpointFrom` should return:
```
https://169.254.170.2/v2/credentials/e7e38a7b-8ed8-4245-8557-d88b02aca7c7
```

Reply:
So, I am a bit hesitant to go that use case because you can have AWS security tokens generating from a service that is not Fargate and returns the same response as Fargate's relative endpoint, 

I do think we should internally, concatenate.

Reply:
Can you be more specific? When does STS do that and assign it to `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`?

Reply:
I can imagine a service that does not have a unique relative url like Fargate but it can generate STS tokens. In that case, I would think `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI` should be empty.

So, the cases are 
A) When AWS_CONTAINER_CREDENTIALS_RELATIVE_URI is empty, then we can assume someone is vending it using STS
B) AWS_CONTAINER_CREDENTIALS_RELATIVE_URI is not empty,  then we should assume fargate